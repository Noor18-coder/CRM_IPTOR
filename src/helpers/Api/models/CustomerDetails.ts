export interface CustomerDetailsDefault {
    businessPartner: number,
    internalName: string,
    type: number,
    name: string,
    addressLine1: string,
    addressLine2: string,
    addressLine3: string,
    addressLine4: string,
    postalCode: string,
    country: string,
    area:string,
    language: string,
    invoiceCustomer: number,
    debtorAddressNumber: number,
    debtor: number,
    invoiceAddressNumber: number,
    currency: string
    isParent: boolean,
    active: boolean,
    numberOfActiveOpportunities: number,
    numberOfInactiveOpportunities: number,
    phone: string,
    subsidiaryEntities:[],
    APP_FROM_IPTOR:[],
    CURRENT_SYSTEM: string,
    CUST_CLASSIFICATION: string,
    CUSTOMER_AT_RISK: boolean,
    DC1_VERSION: [],
    DO_NOT_CONTACT: boolean,
    GLOBAL_PARENT: number,
    IBS_ASW_RELEASE: number,
    INDUSTRY: string,
    OWNER_ID: string,
    PERP_USE_MAINT_CONV: boolean,
    REFERENCEABLE: boolean,
    TYPE:[]   
};

export interface CustomerDetailsParams {
    businessPartner: string
}

export interface CustomerDetailsDefaultResponse {
       data: CustomerDetailsDefault[]  
}


export interface  CustomerDetailsContactsGroupItem {
  contactId: string,
  contactDC: string,
  contactPerson: string,
  email: string,
  phone: number,
  fax: string,
  businessPartner: number,
  role: string,
  ADDRESS: string,
  ADDRESS_2: string,
  DO_NOT_CONTACT: boolean,
  NO_LONGER_AT_COMP: boolean,
  OPTED_OUT_OF_EMAIL: boolean,
  PRIMARY_CONTACT: boolean,
  mobile ?:string,
  linkedin ?:string,
  ACTIVE: boolean
};

export interface CustomerDetailsContactsGroupItemResponse {
    data: CustomerDetailsContactsGroupItem[]
}

export interface CustomerDeleteParams {
    businessPartner: string,
    contactDC: string
}

export interface CustomerDeleteResponse {
    data: CustomerDeleteParams
}

export interface  CrmCountry {
  country: string,
  description: string,
  ISOcountry: string,
  validateState: number,
  validateCounty: number
};

export interface CrmCountryParams {
  country: string
}

export interface  Area {
 area:string,
 description:string
};

export interface AreaParams {
  area: string
}
  export interface  CrmUserDetails {
    user: string,
    description: string,
    status: boolean,
    groupProfile: boolean,
    interactiveProcessingOk: boolean,
    defaultEnquiryStdSystemUnit: string,
    defaultEnquiryStdSystemUnitP: string,
    disPeriodChangeAllowed: boolean,
    handler: string,
    defaultOrderTypeReq: string,
    EMAIL: [],
    MANAGER: string,
    PHONE: [],
    ROLE: string
  }
  
    export interface CrmUserDetailsParams {
      user: string
  }