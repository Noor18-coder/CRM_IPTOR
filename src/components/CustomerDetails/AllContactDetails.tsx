import React from 'react';
import { Card, Image, Accordion } from 'react-bootstrap';
import { Dispatch } from 'redux';
import { useDispatch } from 'react-redux';
import { useMediaQuery } from 'react-responsive';
import * as models from '../../helpers/Api/models';
import ImageConfig from '../../config/ImageConfig';
import {
  setBusinessPartnerWindowActive,
  setBusinessPartnerWindowGroup,
  setBusinessPartnerContactId,
  setBusinessPartnerLoader,
} from '../../store/AddCustomer/Actions';
import CustomerDetailsApi from '../../helpers/Api/CustomerDetailsApi';

interface Props {
  title: string;
  contactData: models.CustomerDetailsContactsGroupItem[];
  customerData: models.CustomerDetailsDefault;
}

export const AllContactsAccordian: React.FC<Props> = ({ title, contactData, customerData }) => {
  const isMobile = useMediaQuery({ maxWidth: 767 });
  const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 991 });
  const [activeClass, setActiveClass] = React.useState('');
  const dispatch: Dispatch<any> = useDispatch();

  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };

  const toggleDrawer = (groupType: string, contactId: string) => {
    document.body.classList.add('body-scroll-hidden');
    dispatch(setBusinessPartnerWindowActive(true));
    dispatch(setBusinessPartnerWindowGroup(groupType));
    dispatch(setBusinessPartnerContactId(contactId));
  };

  const onInputValueChange = (e: React.ChangeEvent<HTMLInputElement>, id: string) => {
    const selectedContact = contactData.filter((element) => element.contactDC === id);
    selectedContact[0].ACTIVE = e.currentTarget.checked;
    dispatch(setBusinessPartnerLoader(true));
    CustomerDetailsApi.updateContactDetails(selectedContact[0]);
    setTimeout(function () {
      window.location.reload(false);
      dispatch(setBusinessPartnerLoader(false));
    }, 3000);
  };

  const deleteContact = (customerId: string, contactId: string) => {
    dispatch(setBusinessPartnerLoader(true));
    CustomerDetailsApi.deleteContact(customerId, contactId);
    setTimeout(function () {
      window.location.reload(false);
      dispatch(setBusinessPartnerLoader(false));
    }, 3000);
  };

  return (
    <Accordion defaultActiveKey="0">
      <Card className="contact-class">
        <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
          <span className="cust-title">{title}</span>
          <span className="cust-info">
            <span>{contactData.length} CONTACTS</span>
          </span>
          {customerData.active ? (
            <Image
              src={ImageConfig.ADD_BTN}
              className="add-img action-icon"
              alt="Add"
              title="Add"
              onClick={() => toggleDrawer('add contact fields', '')}
            />
          ) : null}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <div className="accr-body-container customers-comp">
            <div className="container-fluid">
              <div className="row">
                {contactData.length ? (
                  contactData.map((obj: models.CustomerDetailsContactsGroupItem) => {
                    return (
                      <div className="col-md-4">
                        <Card className="accordian-card mb-4">
                          <Card.Body className={!obj.ACTIVE ? 'disabled-card' : ''}>
                            <div className="name-add-sec">
                              <p className="person-name-desig">
                                <span className="name">{obj.contactPerson}</span>
                                <span className="designation">{obj.role}</span>
                              </p>
                              <p className="mailid">{obj.email ? obj.email : '--'}</p>
                              <p className="contact-num">{obj.phone ? obj.phone : '--'}</p> <br />
                              <p className="contact-address">
                                <span className="line1">{obj.ADDRESS ? obj.ADDRESS : '--'}</span>
                              </p>
                            </div>
                            <div className="card-action-icons">
                              {obj.ACTIVE && (
                                <>
                                  <Image
                                    className="card-delete action-icon"
                                    height="20"
                                    src={isMobile || isTablet ? ImageConfig.DELETE_ICON : ImageConfig.CLOSE_BTN}
                                    alt="Delete"
                                    title="Delete"
                                    onClick={() =>
                                      deleteContact(
                                        obj.businessPartner ? obj.businessPartner.toString() : '',
                                        obj.contactDC ? obj.contactDC.toString() : ''
                                      )
                                    }
                                  />
                                  <Image
                                    src={ImageConfig.EDIT_ICON}
                                    className="action-icon"
                                    alt="Edit"
                                    title="Edit"
                                    onClick={() => toggleDrawer('contact fields', obj.contactDC ? obj.contactDC.toString() : '')}
                                  />
                                </>
                              )}
                            </div>
                            <div className="active-switch">
                              <span className="active-text">
                                Active
                                <label className="switch active-box">
                                  <input
                                    type="checkbox"
                                    id="ACTIVE"
                                    checked={obj.ACTIVE}
                                    onChange={(e) => onInputValueChange(e, obj.contactDC ? obj.contactDC.toString() : '')}
                                  />
                                  <span className="slider round">&nbsp;</span>
                                </label>
                              </span>
                            </div>
                          </Card.Body>
                        </Card>
                      </div>
                    );
                  })
                ) : (
                  <div className="padding-28"> No Contacts Found </div>
                )}
              </div>
            </div>
          </div>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};
