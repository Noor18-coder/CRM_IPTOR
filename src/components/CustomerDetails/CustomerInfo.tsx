import React from 'react'
import { useMediaQuery } from 'react-responsive';
import { Dispatch } from "redux";
import { useSelector, useDispatch } from "react-redux";
import { setOpportunityWindowActive } from '../../store/AddOpportunity/Actions';
import ImageConfig from '../../config/ImageConfig';
import { Image } from 'react-bootstrap';
import { NavSection } from '../Shared/DetailsNav/NavSection';
import { useHistory } from "react-router-dom";
import { CustomerDetailsDefault, CrmCountry } from '../../helpers/Api/models';
import CustomerDetailsApi from '../../helpers/Api/CustomerDetailsApi';
import * as models from '../../helpers/Api/models';
import { AppState } from "../../store/store";
import AddCustomerApi from '../../helpers/Api/AddCustomer';
import { setBusinessPartnerLoader } from '../../store/AddCustomer/Actions';

export interface Data {
  data: CustomerDetailsDefault
}

const CustomerInfo: React.FC<Data> = (props) => {
  const isMobile = useMediaQuery({ maxWidth: 767 });
  const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 991 })
  const state: AppState = useSelector((state: AppState) => state);
  const history = useHistory();
  const backToOpportunityList = () => {
    history.goBack()
  }
  const [country, setCountry] = React.useState<models.CrmCountry[]>([]);
  const [customerFields, setCustomerFields] = React.useState<any>({});
  const dispatch: Dispatch<any> = useDispatch();

  const toggleDrawer = (event:React.MouseEvent<HTMLElement> | React.KeyboardEvent) => {
    dispatch(setOpportunityWindowActive(true));
  };

    const onInputValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        let customerData = { ...customerFields }
        customerData.active = e.currentTarget.checked
        setCustomerFields(customerData)
        AddCustomerApi.update(customerData);
        dispatch(setBusinessPartnerLoader(true));
        setTimeout(function () {
            window.location.reload(false);
            dispatch(setBusinessPartnerLoader(false));
        }, 3000);
  }

  React.useEffect(() => {
    CustomerDetailsApi.getCountry(props.data.country).then((data) => {
      setCountry(data);
    });
      setCustomerFields(props.data)
  }, []);
  return (
    <>
      {(isMobile || isTablet) ?
        <section className="customer-mobilecard">
          <div className="d-flex justify-content-between customer-row">
            <div className="lft-custname" onClick={backToOpportunityList}>
              <p>{props.data.name}<span className="location">{country ? country.map((data: CrmCountry) => {
                return data.description
              }) : null}</span></p>
            </div>
            <div className="rgt-actioncol">
              <ul className="list-inline ">
                <li className="list-inline-item"><img src={ImageConfig.HISTORY} alt="History" title="History" /></li>
                <li className="list-inline-item"><img src={ImageConfig.STAR} alt="Star" title="Star" /></li>
                <li className="list-inline-item"><img src={ImageConfig.MORE_V_ELLIPSIS} alt="More" title="More" /></li>
              </ul>
            </div>
          </div>

          <div className="customer-details d-flex justify-content-between">
            <div className="left-data">
              <p><span>Active</span></p>
            </div>

            <div className="right-data">
              <p><span><label className="switch">
                <input type="checkbox" checked={props.data.active} />
                <span className="slider round"></span>
              </label></span></p>
            </div>
          </div>
          <div className="customer-details d-flex justify-content-between">
            <div className="left-data">
              <p><span>Parent Group</span></p>
            </div>

            <div className="right-data">
              <p><span><label className="switch">
                <input type="checkbox" checked={props.data.isParent} />
                <span className="slider round"></span>
              </label></span></p>
            </div>
          </div>

          <div className="customer-details d-flex justify-content-between">
            <div className="left-data">
              <p><span>New Opportunity</span></p>
            </div>

            <div className="right-data">
              <p><span> <Image src={ImageConfig.ADD_ICON} /></span></p>
            </div>
          </div>
        </section> :
        <>
          <NavSection backToOpportunityList={backToOpportunityList} />
          <section className="d-flex justify-content-between sec-customer-addr">
            <div className="cust-name">
              <p>{props.data.name}<span>{country ? country.map((data: CrmCountry) => {
                return data.description
              }) : null}</span></p>
            </div>
            <div className="mid-sec">
              <ul className="list-inline">
                <li className="list-inline-item"> <span>Active <label className="switch">
                  <input type="checkbox" checked={customerFields.active} onChange={!!state.auth.user.role && state.auth.user.role === 'Admin' ? onInputValueChange : undefined} />
                  <span className={!!state.auth.user.role && state.auth.user.role === 'Admin' ? 'slider round' : 'slider round disabled-checkbox'}></span>
                </label> </span> </li>
                <li className="list-inline-item"><span className="p-left">Parent Group <label className="switch">
                  <input type="checkbox" checked={props.data.isParent} />
                  <span className="slider round"></span>
                </label> </span></li>
              </ul>
            </div>

            <div className="sec-add-cust ">
              <button className="btn add-customer" onClick={toggleDrawer} >+ New Opportunity</button>
            </div>
          </section>
        </>
      }
    </>
  )
}
export default CustomerInfo
