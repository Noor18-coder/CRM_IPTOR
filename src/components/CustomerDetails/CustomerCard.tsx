import React from 'react';
import { Dispatch } from "redux";
import { useDispatch, useSelector } from "react-redux";
import { useHistory } from 'react-router';
import { Accordion, Card, Image } from 'react-bootstrap';
import { useMediaQuery } from 'react-responsive';
import i18n from '../../i18n'

import ImageConfig from '../../config/ImageConfig';
import { AppState } from "../../store/store";
import { AllContactsAccordian } from './AllContactDetails';
import { MoreInfoAccordian, MoreInfoAccordianMobile } from './MoreInfo'
import * as models from '../../helpers/Api/models';
import { CustomerDetailsDefault, CustomerDetailsContactsGroupItem, Area } from  '../../helpers/Api/models';
import CustomerDetailsApi from '../../helpers/Api/CustomerDetailsApi';
import OpportunityDetailsApi from '../../helpers/Api/OpportunityDetailsApi';

import Container from '../AddCustomer/Container';
import { setBusinessPartnerWindowActive, setBusinessPartnerLoader, setBusinessPartnerWindowGroup } from '../../store/AddCustomer/Actions';

export interface Data {
  data:CustomerDetailsDefault,
  contactsData:CustomerDetailsContactsGroupItem[]
}

const CustomerCard:React.FC<Data> = (props) =>   {
  const isMobile = useMediaQuery({ maxWidth: 767 });
  const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 991 })
  const [subsidiaryEntities, setsubsidiaryEntities] = React.useState<models.CustomerDetailsDefault[]>([]);
  const [ownerDetails, setOwnerDetails] = React.useState<models.CrmUserDetails>();
  const numberOfInactiveOpportunities = props.data.numberOfInactiveOpportunities? props.data.numberOfInactiveOpportunities :0;
  const numberOfActiveOpportunities = props.data.numberOfActiveOpportunities ? props.data.numberOfActiveOpportunities :0;
  const [area, setArea] = React.useState<models.Area[]>([]);
  const [customerMoreInfoGroup, setCustomerMoreInfoGroup] = React.useState<models.IStringList>();

  const dispatch: Dispatch<any> = useDispatch();
  const history = useHistory();
  const state: AppState = useSelector((state: AppState) => state);
  const customerAttributes = state.enviornmentConfigs.customerAttributes

  React.useEffect(() => {      
    CustomerDetailsApi.getOwnerDetailsByName(props.data.OWNER_ID).then((data) => {
            setOwnerDetails(data);
    });
    if(props.data.subsidiaryEntities) {
      Promise.all(props.data.subsidiaryEntities.map((id: any) => {
        return CustomerDetailsApi.get(id);
      })).then((entityData) => {
        setsubsidiaryEntities(entityData);
        return entityData;
      });
    }

    CustomerDetailsApi.getArea(props.data.area).then((data) => {
      setArea(data);
    });

  OpportunityDetailsApi.getCustomerGroupInfo(props.data.businessPartner.toString()).then((data) => {
        customerAttributes.map((item) => {
            if (!data.some(ele => ele.attributeType === item.attributeType)) {
                data.push({ attributeType: item.attributeType, group: item.group, attributeValue: '', description: item.description });
            }
        });
        const groups = new Set(data.map((obj) => { return obj.group }));
        let response: any = {};
        groups.forEach((group: string) => {
            const groupName = group.toLowerCase();
            response[groupName] = data.filter((obj) => obj.group.toLowerCase() === groupName);
        });
        setCustomerMoreInfoGroup(response);
  });
      dispatch(setBusinessPartnerLoader(false));
      dispatch(setBusinessPartnerWindowActive(false));
  }, []);

  const [activeClass , setActiveClass] = React.useState("");
  const toggleAccordion = () => {
    setActiveClass(activeClass === "" ? "active" : "");
    }

    const toggleDrawer = (open: boolean, groupType: string) => (event: React.MouseEvent<HTMLElement> | React.KeyboardEvent) => {
        document.body.classList.add('body-scroll-hidden');
        dispatch(setBusinessPartnerWindowActive(true));
        dispatch(setBusinessPartnerWindowGroup(groupType));
    };

    const openOpptyList = (flag: boolean) => (event: React.MouseEvent<HTMLElement> | React.KeyboardEvent) => {
        history.push({ pathname: "/opportunities", state: { selectCustomer: props.data.businessPartner, activeOp: flag } })
    }

    return (
      <>
        <section className="sec-info-accordion">
          <Accordion  defaultActiveKey="0">
              <Card className="cust-details">
                <Accordion.Toggle className={activeClass} onClick={toggleAccordion}  as={Card.Link} eventKey="1">
                    {i18n.t('basicInfo')}
                </Accordion.Toggle>
                <Accordion.Collapse eventKey="1">
                  <Card.Body className="accr-body-container">  
                     <div className="group-icon"><Image src={ImageConfig.EDIT_ICON} className="action-icon" alt="Edit" title="Edit" onClick={toggleDrawer(true, 'default fields')} /></div>
                     <ul className="list-inline bdy-list-item accr-list-columns">
                          <li className={(isMobile || isTablet) ? '' : "list-inline-item"}>
                            <span>Contact Address</span>{props.data.addressLine1}<p>&nbsp;</p>
                          </li>
                          <li className={(isMobile || isTablet) ? '' : "list-inline-item"}>
                              <span>Region</span>{area ? area.map((data: Area) => {
                              return data.description
                              }) : '--'}<p>&nbsp;</p>
                          </li>
                          <li className={(isMobile || isTablet) ? '' : "list-inline-item"}>
                            <span>Phone Number</span>{props.data.phone? props.data.phone : '--'}<p>&nbsp;</p>
                          </li>
                     </ul>
                  </Card.Body>
                </Accordion.Collapse>
              </Card>
           </Accordion>
           <Container containerType='edit' containerData={props} groupType={state.addBusinessPartner.businessPartnerWindowGroup} contactId={state.addBusinessPartner.businessPartnerContactId} />
        </section>

        <section className="sec-info-accordion">
                {(isMobile || isTablet) ?
                    customerMoreInfoGroup && <MoreInfoAccordianMobile title={i18n.t('moreInfo')} data={customerMoreInfoGroup} /> :
                    customerMoreInfoGroup && <MoreInfoAccordian title={i18n.t('moreInfo')} data={customerMoreInfoGroup} />}
        </section>


      { (isMobile || isTablet) ? 
      <section className="customer-mobilecard">
          <div className="customer-details d-flex justify-content-between">
            <div className="lft-col">
                Account owner
               <span>
               {ownerDetails && ownerDetails.user? ownerDetails.user : '--'}
             </span>
            </div>
        
            <div className="right-data">
              {/*<p><Image className="edit" src={ImageConfig.EDIT_ICON} alt="edit" title="edit" /></p>*/}
            </div>
          </div>
          <div className="customer-details d-flex justify-content-between">
            <div className="lft-col">
               <span>
             {ownerDetails && ownerDetails.PHONE ? ownerDetails.PHONE : '--'}
             </span>
            </div>

            <div className="right-data">
              <p><Image  width="25" src={ImageConfig.PHONE} alt="phone" title="phone" /></p>
            </div>
          </div>

          <div className="customer-details d-flex justify-content-between">

            <div className="lft-col">
               <span>
              {ownerDetails && ownerDetails.EMAIL ? ownerDetails.EMAIL : '--'}
             </span>
            </div>

            <div className="right-data">
              <p><Image  width="25" src={ImageConfig.MAIL} alt="mail" title="mail" /></p>
            </div>
          </div>
      </section>:
      <section className="d-flex justify-content-between sec-customer-desc">
           <div className="accr-body-container">  
               <ul className="list-inline bdy-list-item owner-section">
               <li className="list-inline-item"><span>Account owner</span> {ownerDetails && ownerDetails.user ? ownerDetails.user : '--'}</li>
               <li className="list-inline-item"><Image  width="25" src={ImageConfig.PHONE} alt="phone" title="phone" />&nbsp;&nbsp;{ownerDetails && ownerDetails.PHONE ? ownerDetails.PHONE : '--'}</li>
               <li className="list-inline-item"><Image  width="25" src={ImageConfig.MAIL} alt="mail" title="mail" /> &nbsp;&nbsp;{ownerDetails && ownerDetails.PHONE ? ownerDetails.EMAIL : '--'}</li>
             </ul></div>
       </section> }

       <section className="d-flex sec-customer-desc">
            <div className="cust-group">
              <p>{i18n.t('appFromIptor')}</p>
            </div>
            {props.data.APP_FROM_IPTOR?
            props.data.APP_FROM_IPTOR.map((name: any) => {
            return ( <div className="group-sec">
              <ul className="list-inline">
              <li className="list-inline-item"><span className="cust-info">{name}</span> </li>
              </ul>
            </div>)
              }):null}
      </section>

      <section className="sec-info-accordion">
        <AllContactsAccordian title={' All Contact'} contactData={props.contactsData} />
      </section>

      <section className="d-flex sec-customer-desc">
        <div className="cust-group">
          <p>Opportunities</p>
        </div>
        <div className="group-sec">
          <ul className="list-inline">
          <li className="list-inline-item"><span className={numberOfActiveOpportunities !== 0 ? "cust-info action-icon" : "cust-info"} onClick={numberOfActiveOpportunities !== 0 ? openOpptyList(true) : undefined}>{numberOfActiveOpportunities} ACTIVE</span> </li>
          <li className="list-inline-item"><span className={numberOfInactiveOpportunities !== 0 ? "cust-info action-icon" : "cust-info"} onClick={numberOfInactiveOpportunities !== 0 ? openOpptyList(false) : undefined}>{numberOfInactiveOpportunities} CLOSED</span></li>
          <li className="list-inline-item"><span className="cust-info">{numberOfActiveOpportunities + numberOfInactiveOpportunities} TOTAL</span></li>
          </ul>
        </div>
      </section>

      <section className="d-flex sec-customer-desc">
        <div className="cust-group">
          <p>Subsidiary - Entity</p>
        </div>
        {subsidiaryEntities? subsidiaryEntities.map((data: CustomerDetailsDefault) => {
        return (<div className="group-sec">
          <ul className="list-inline">
          <li className="list-inline-item"><span className="cust-info">{data.name +' - ' +data.area}</span> </li>
          </ul>
        </div>)
          }):null}
    
        {/* <div className="sec-status">
        <ul className="list-inline"><li className="list-inline-item">
        <Image src={ImageConfig.ADD_BTN} alt="Add" title="Add" /> 
        <Image className="del-icon" src={ImageConfig.EDIT_ICON} alt="edit" title="edit" /></li></ul>
        </div> */}
      </section>
      
      </>
    )
}

export default CustomerCard
