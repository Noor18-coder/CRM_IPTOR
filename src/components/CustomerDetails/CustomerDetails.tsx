import React from 'react'
import Header from '../Shared/Header/Header'
import Footer from '../Shared/Footer/Footer'
import CustomerCard from './CustomerCard'
import CustomerInfo from './CustomerInfo'
import * as models from '../../helpers/Api/models';
import CustomerDetailsApi from '../../helpers/Api/CustomerDetailsApi';

const CustomerDetails: React.FC = (props: any) => {
    const custId = props.location.state.custId;
    const [defaultCustDetail, setDefaultCustDetails] = React.useState<models.CustomerDetailsDefault>();
    const [contactDetails, setContactDetails] = React.useState<models.CustomerDetailsContactsGroupItem[]>([]);

    React.useEffect(() => {
        CustomerDetailsApi.get(custId).then((data) => {
            setDefaultCustDetails(data);
        });
        CustomerDetailsApi.getAllContactDetails(custId).then((data) => {
            setContactDetails(data);
        });

},[]);
        return (
            <>
            <Header page={2}/>
            <section className="main-wrapper customer">
            <div className="container-fluid">
                {defaultCustDetail ? <CustomerInfo data={defaultCustDetail} /> : null}
                {defaultCustDetail ? <CustomerCard data={defaultCustDetail} contactsData={contactDetails}/> : null}
            </div>
            </section>
            <Footer />
            </>
        
        )
}

export default CustomerDetails;