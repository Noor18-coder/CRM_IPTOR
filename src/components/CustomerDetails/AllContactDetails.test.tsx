import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { AllContactsAccordian } from './AllContactDetails';
import StoreMock from '../../mocks/Store.mock';
import CustomerDetails from '../../mocks/CustomerDetails.mock';

const store = StoreMock.createInitialStore();
const contactData = CustomerDetails.getCustomerDetailsContactsGroupItemConstVal();

describe('[CustomerDetails] AllContactsAccordian', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AllContactsAccordian contactData={contactData} title="title" activeCustomer={false} customerOwner="" />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
