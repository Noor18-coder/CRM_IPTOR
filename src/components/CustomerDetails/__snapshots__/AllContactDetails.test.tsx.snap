// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CustomerDetails] AllContactsAccordian should renders correctly 1`] = `
<div
  className="accordion"
>
  <div
    className="contact-class card"
  >
    <a
      className="card-link"
      onClick={[Function]}
    >
      <span
        className="cust-title"
      >
        title
      </span>
      <span
        className="cust-info"
      >
        <span>
          1
           CONTACTS
        </span>
      </span>
      <img
        alt="Add"
        className="add-img action-icon"
        onClick={[Function]}
        src="add_btn.svg"
        title="Add"
      />
    </a>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="accr-body-container customers-comp"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4"
            >
              <div
                className="accordian-card mb-4 card"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="name-add-sec"
                  >
                    <p
                      className="person-name-desig"
                    >
                      <span
                        className="name"
                      >
                        New
                      </span>
                    </p>
                    <p
                      className="mailid"
                    >
                      Plastic
                    </p>
                    <p
                      className="contact-num"
                    >
                      5130757854
                    </p>
                     
                    <br />
                    <p
                      className="contact-address"
                    >
                      <span
                        className="line1"
                      >
                        Profound
                      </span>
                    </p>
                  </div>
                  <div
                    className="card-action-icons"
                  >
                    <img
                      alt="Delete"
                      className="card-delete action-icon"
                      height="20"
                      onClick={[Function]}
                      src="close.svg"
                      title="Delete"
                    />
                    <img
                      alt="Edit"
                      className="action-icon"
                      onClick={[Function]}
                      src="edit-icon.svg"
                      title="Edit"
                    />
                  </div>
                  <div
                    className="active-switch"
                  >
                    <span
                      className="active-text"
                    >
                      Active
                      <label
                        className="switch active-box"
                      >
                        <input
                          checked={true}
                          disabled={false}
                          id="ACTIVE"
                          onChange={[Function]}
                          tabIndex={0}
                          type="checkbox"
                        />
                        <span
                          className="slider round"
                        >
                          Â 
                        </span>
                      </label>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
